pkg/resources/management.cattle.io/v3/projectroletemplatebinding/validator.go

```diff
@@ -60,23 +62,26 @@ func (v *Validator) ValidatingWebhook(clientConfig admissionregistrationv1.Webho
 	return []admissionregistrationv1.ValidatingWebhook{*admission.NewDefaultValidatingWebhook(v, clientConfig, admissionregistrationv1.NamespacedScope, v.Operations())}
 }
 
-// Admit is the entrypoint for the validator. Admit will return an error if it unable to process the request.
-// If this function is called without NewValidator(..) calls will panic.
+// Admit is the entrypoint for the validator. Admit will return an error if it's unable to process the request.
+// If this method is called on a nil Validator, it panics.
 func (v *Validator) Admit(request *admission.Request) (*admissionv1.AdmissionResponse, error) {
 	listTrace := trace.New("projectRoleTemplateBindingValidator Admit", trace.Field{Key: "user", Value: request.UserInfo.Username})
 	defer listTrace.LogIfLong(admission.SlowTraceDuration)
 
+	fieldPath := field.NewPath("projectroletemplatebinding")
+	var fieldErr *field.Error
+
 	if request.Operation == admissionv1.Update {
 		oldPRTB, newPRTB, err := objectsv3.ProjectRoleTemplateBindingOldAndNewFromRequest(&request.AdmissionRequest)
 		if err != nil {
 			return nil, fmt.Errorf("failed to decode PRTB objects from request: %w", err)
 		}
 
-		if err = validateUpdateFields(oldPRTB, newPRTB); err != nil {
+		if fieldErr = validateUpdateFields(oldPRTB, newPRTB, fieldPath); fieldErr != nil {
 			return &admissionv1.AdmissionResponse{
 				Result: &metav1.Status{
 					Status:  "Failure",
-					Message: err.Error(),
+					Message: fieldErr.Error(),
 					Reason:  metav1.StatusReasonBadRequest,
 					Code:    http.StatusBadRequest,
```

Comment:
Looking at the [validator.go](https://github.com/rancher/rancher/blob/1f014b7fcd734e9321555b613cc28b1b12bf8eeb/pkg/api/norman/customization/roletemplatebinding/validator.go#L77) from Norman, the various errors have specific error codes. @KevinJoiner Should we be passing these along to the admission control response?

Reply:
We don't need to follow Norman errors, but we should take a look at using specific http error codes.